src/alfasim_sdk/alfasim_sdk_api/api.h

```diff
@@ -812,6 +812,51 @@ DLL_EXPORT int get_liq_liq_flow_pattern_input_variable(
     void* ctx, double* out, const char* var_name, int phase_id
 );
 
+/*!
+    Gets the wall names and wall properties for a given control volume. The wall names will be given
+    as an array of char pointers and the properties by an array data pointer. This method also provide the
+    size of the given arrays.
+
+    IMPORTANT:
+    The given wall can be a fluid or solid, be aware of this when you start to use the properties.
+    List of fluid properties:
+    - Volumetric Thermal Expansion
+    - Dynamic Viscosity
+
+    List of `prop_name`:
+    - `"layer_thickness"`: Wall Layer Thickness [m]
+    - `"rho"`: Wall Density [kg/m3]
+    - `"cp"`: Wall Specific Heat Capacity [J/kg.K]
+    - `"volumetric_thermal_expansion"`: Wall Volumetric Thermal Expansion [1/K]
+    - `"cP"`: Wall dynamic viscosity [cP]
+    - `"k"` : Wall Thermal Conductivity [W/m.degC]
+    - `"layer_is_fluid"`: Informs if the wall is a fluid [1 - is fluid; 0 - is not fluid]
+
+    Example of usage:
+
+                                                 [prop_wall_4]
+         [prop_wall_3]                           [prop_wall_3]
+         [prop_wall_2]       [prop_wall_2]       [prop_wall_2]
+         [prop_wall_1]       [prop_wall_1]       [prop_wall_1]
+         [prop_wall_0]       [prop_wall_0]       [prop_wall_0]
+               |                   |                   |
+    ---[control_volume_1]--[control_volume_2]--[control_volume_3]-->  (Pipe)
+
+    ~~~~~{.cpp}
+    errcode = get_wall_properties(
+        ctx, &wall_names, &prop_values, "layer_thickness", control_volume_id, &size_wall);
+    ~~~~~
+
+    @param[in] ctx ALFAsim's plugins context.
+    @param[out] wall_names Wall names values array.
+    @param[out] prop_values Wall properties values array.
+    @param[in] prop_name String with the property name. See the list of possible values above.
+    @param[in] control_volume Control volume id.
+    @param[out] size Size of the `wall_names` and `prop_values` array of values.
+    @return An #error_code value.
+*/
+DLL_EXPORT int get_wall_properties(void* ctx, char*** wall_names, void** prop_values, const char* prop_name, int control_volume, int* size);
```

Comment:
DLL_EXPORT int get_wall_properties(void* ctx, char*** wall_names, void** prop_values, const char* prop_name, int control_volume, int* size);

I am against the api returning pre allocated buffers (the caller should allocate those).
But someone put those kind of stuff in the api already... and someone could argue that praticatily is best (I disagree here).

But having an "out" argument as `void *` is just not OK.
`ctx` is a `void *` because the caller is not supposed to type cast it (alfasim supply that as an opaque pointer).
I will not bend on this just because there are examples of that in the api. I don't think those should be there.

Create variations with the correct explicit type, and if called to obtain a property in the wront type return a failure error core.
```c
DLL_EXPORT int get_wall_properties_bool(void* ctx, char*** wall_names, bool** prop_values, const char* prop_name, int control_volume, int* size);
DLL_EXPORT int get_wall_properties_double(void* ctx, char*** wall_names, double** prop_values, const char* prop_name, int control_volume, int* size);
...
```
(my C is rusty and I don't remember if/how aligiment affect "bool" arrays)

Reply:
Check existing [error codes](https://github.com/ESSS/alfasim-sdk/blob/dbf8d6513bf544cf4fa683ff01ccd86b15515c9a/src/alfasim_sdk/alfasim_sdk_api/common.h#L9-L20).

Maybe:
- UNDEFINED_DATA for when the user ask for an invalid `prop_name` (btw, are `prop_name` fixed? If so maybe we should have one specific function for each property);
- OUT_OF_BOUNDS for invalid `control_volume` (but arguably UNDEFINED_DATA  also couldbe used I guess);

See BUFFER_SIZE_INSUFFICIENT in the list (with an near 0 number)? This is because it was planned to the buffers to be allocated by the caller.

Reply:
Also.
By `wall_names`. Do you mean `material_names`? 

I ask that because a point in a pipe can have just one wall, one wall can have multiple (unnamed layers), and every layer have one material (named).
And a wall could have multiple layers with the same material.

Reply:
@prusse-martin perhaps we could discuss it in a call. I don't remember the details, but this definition of outputs on API functions was discussed a lot before Tarcisio and William implemented it in this way.

Reply:
hmmm, ok

- Regarding the `void* ` for `out` it seems fair to create two separate methods. 

- About the error_code, thanks for the suggestions, I will make those changes (your ideas are much better)

- `prop_name ` can be a set of 7 names, but I think that having one specific function for each property could lead to some code replication. 

> By wall_names. Do you mean material_names?
Actually, it suppose to be `material_names`. I decided to export the names considering that the user could want to know the material's name to take some action (maybe use predefined values for other property). But we can think about create a different method to return that. (`get_wall_material_names`)

Reply:
> ... export the names considering that the user could want to know the material's name to take some action...

Those are user inputs not fixed names, they could be in English, Portuguese, Russian, Mandarin...
We should not guess what we are dealing with, forsimulation purposes the name should not be relevant and use just the physical properties of the material.

Reply:
I applied all the changes we talked about in the meeting

- [ ] split get_wall_properties into get_wall_properties, get_wall_names and get_wall_check_for_fluid_layer
- [ ] use OUT_OF_BOUNDS instead of UNDEFINED_DATA 

I also checked with alfatwin if get_wall_names is needed. The answer was yes.